src/main/java/apoc/graph/inverse/builder/IdBuilder.java

```diff
@@ -0,0 +1,49 @@
+package apoc.graph.inverse.builder;
+
+import apoc.graph.inverse.util.Util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class IdBuilder {
+
+    public static final String ID = "id";
+    public static final String TYPE = "type";
+
+    private Util id;
+    private Util type;
+    private Map<String, String> fields;
+
+    public IdBuilder() { }
+
+    public IdBuilder buildId(Map<String, Object> obj) {
+        Object typeObj = obj.get(IdBuilder.TYPE);
+        Object idObj = obj.get(IdBuilder.ID);
+
+        if (idObj == null) {
```

Comment:
check and report both best for the whole doc
report all issues in one go

optional: validation procedure